core/services/log/registrations.go

```diff
@@ -61,47 +63,50 @@ func newRegistrations() *registrations {
 	return &registrations{
 		subscribers: make(map[uint64]*subscribers),
 		decoders:    make(map[common.Address]ParseLogFunc),
+
+		// a high value for lowestNumConfirmations is set until first subscriber,
+		// to avoid a special case in comparison to its NumConfirmations
+		lowestNumConfirmations: uint64(math.MaxUint64),
```

Comment:
```suggestion
		lowestNumConfirmations: math.MaxUint64,
```

Reply:
Alternatively use an `int64` and `-1` as the sentinel value, or even a nullable pointer

Reply:
removed this field now, it wasn't actually needed